src/Loading/errors.ts

```diff
@@ -0,0 +1,19 @@
+import { BaseError } from "../Misc/baseError";
+import { ErrorCodes } from "../Misc/errorCodes";
+
+/**
+ * Application error to indicate load scene failures.
+ */
+export class SceneLoaderError extends BaseError {
```

Comment:
This seems inconsistent with other usages. Why does only SceneLoaderError have a special derived error but other classes (like glTF loader) don't? I see it auto sets the SceneLoaderError code, but I'm not sure on the error hierarchy.

Reply:
SceneLoader would also wrap an inner exception unlike the gltf loader error. The innerEx would have the actual error code. The error hierarchy does get confusing :/. Totally makes sense to remove this class since we would be also checking for the inner exception. In which case, BaseError would wrap a BaseError/Error.

Reply:
Having an additional layer here would allow people to do something like `if (error instanceof SceneLoaderError)` but because there is already an error code with categories, it's not really useful anymore. I think the general pattern should be to introduce a new hierarchy only if there are additional properties.